protocol/0093-PAPU-protocol_automated_purchases.md

```diff
@@ -0,0 +1,36 @@
+# Protocol Automated Purchases
+
+In order to allow the protocol to better manage collected fees a framework for regular, fully automated exchanges of one token for another on the protocol's spot markets should be implemented. In order to allow the protocol to receive a fair rate for the token exchange, and taking into account that not all spot markets will have deep liquidity, this should be handled through use of a scheduled auction.
+
+This auction should have several defining features:
+
+ - It should be triggerable on a defined schedule
+ - An event should be emitted at a known time in advance of the auction announcing the amount to be exchanged
+ - The price at which the protocol is willing to exchange should be configurable as some offset to a predefined set of oracle inputs
+
+## Automated Purchase Configuration
+
+An automated purchase program is configured through a number of parameters:
+
+ - **Token From**: The source token which will be sold
+ - **Account Type From**: The account type for the network from which the tokens will be sourced
+ - **Account Type To**: The account type for the network to which the purchased tokens will be sent
+ - **Market ID**: The market which will be used to enact the purchase/sale
+ - **Price Oracle**: The oracle which will define an approximate acceptable price for the transaction
+ - **Oracle Offset Factor**: The final acceptable price should be the **Price Oracle** value weighted by this factor (e.g. `1.05` to allow for 5% slippage on the purcase)
```

Comment:
More common use case would be `0.95` so you're letting the protocol for less than the "current price" but if in the auction participants offer more then the auction mechanism should be ensuring you do sell for more. @Jiajia-Cui to scenario test this 🙏 .

Reply:
I can confirm that the currently the auction uncrosses at 100 for the scenario above, FT is [here](https://github.com/vegaprotocol/vega/pull/11651/files)

Reply:
however, the FT has covered a bit more complicated case:

- p2 places a buy order of size 1 at 100 
- p3 places a buy order of size 9 at 107
- and p1 places a sell order of size 10 at 95

the auction uncrosses at 97.5, which is using the earliest uncrossing price (100 and 95), instead of the best uncrossing price (107 and 95)